File: eAnimalShelter/eAnimalShelter.Model/Requests/AdoptionRequestInsertRequest.cs
namespace eAnimalShelter.Model.Requests
{
    public class AdoptionRequestInsertRequest
    {
        public int AnimalId { get; set; }

        public string HousingType { get; set; } = string.Empty;

        public string ExperienceWithPets { get; set; } = string.Empty;

        public int HouseholdMembers { get; set; }

        public string? AdditionalNotes { get; set; }
    }
}

File: eAnimalShelter/UI/eanimalshelter_mobile/lib/models/adoption_request_insert_request.dart
class AdoptionRequestInsertRequest {
  int animalId;
  String housingType;
  String experienceWithPets;
  int householdMembers;
  String? additionalNotes;

  AdoptionRequestInsertRequest({
    required this.animalId,
    required this.housingType,
    required this.experienceWithPets,
    required this.householdMembers,
    this.additionalNotes,
  });

  Map<String, dynamic> toJson() {
    return {
      "animalId": animalId,
      "housingType": housingType,
      "experienceWithPets": experienceWithPets,
      "householdMembers": householdMembers,
      "additionalNotes": additionalNotes,
    };
  }
}